runtime/runtime.go

```diff
@@ -349,6 +357,118 @@ func (r *interpreterRuntime) newAuthAccountValue(
 	)
 }
 
+// InvokeContractFunction invokes a contract function with the given arguments
+// If the contract function accepts an AuthAccount as a parameter the corresponding 
+func (r *interpreterRuntime) InvokeContractFunction(
+	contractLocation common.AddressLocation,
+	functionName string,
+	arguments []interpreter.Value,
+	argumentTypes []sema.Type,
+	context Context,
+) error {
+	context.InitializeCodesAndPrograms()
+
+	runtimeStorage := newRuntimeStorage(context.Interface)
+
+	var interpreterOptions []interpreter.Option
+	var checkerOptions []sema.Option
+
+	functions := r.standardLibraryFunctions(
+		context,
+		runtimeStorage,
+		interpreterOptions,
+		checkerOptions,
+	)
+
+	// create interpreter
+	_, inter, err := r.interpret(
+		nil,
+		context,
+		runtimeStorage,
+		functions,
+		stdlib.BuiltinValues,
+		interpreterOptions,
+		checkerOptions,
+		nil,
+	)
+	if err != nil {
+		return newError(err, context)
+	}
+
+	// ensure the contract is loaded
+	inter = inter.EnsureLoaded(contractLocation)
+
+	// convert addresses to auth accounts
+	// this is here so there is no need to construct an auth account value for the caller of this function
+	for i, argumentType := range argumentTypes {
+		if argumentType != sema.AuthAccountType {
+			// only convert arguments that are supposed to be an auth account
+			continue
+		}
+		if a, ok := arguments[i].(interpreter.AddressValue); ok {
+			// only convert them if they are an address
+			arguments[i] = r.newAuthAccountValue(
+				interpreter.NewAddressValue(a.ToAddress()),
+				context,
+				runtimeStorage,
+				interpreterOptions,
+				checkerOptions,
+			)
+		}
+	}
+
+	contractGlobal, ok := inter.Globals[contractLocation.Name]
+	if !ok {
+		return interpreter.NotDeclaredError{
+			ExpectedKind: common.DeclarationKindContract,
+			Name:         contractLocation.Name,
+		}
+	}
+
+	// get contract value
+	contractValue, ok := contractGlobal.GetValue().(*interpreter.CompositeValue)
+	if !ok {
+		return interpreter.NotDeclaredError{
+			ExpectedKind: common.DeclarationKindContract,
+			Name:         contractLocation.Name,
+		}
+	}
+
+	// prepare invocation
+	invocation := interpreter.Invocation{
+		Self:          contractValue,
+		Arguments:     arguments,
+		ArgumentTypes: argumentTypes,
+		TypeParameterTypes: nil,
+		GetLocationRange: func() interpreter.LocationRange {
+			return interpreter.LocationRange{
+				Location: context.Location,
+			}
+		},
+		Interpreter: inter,
+	}
+
+	contractMember := contractValue.GetMember(inter, func() interpreter.LocationRange {
+		return interpreter.LocationRange{
+			Location: context.Location,
+		}
+	}, functionName)
+
+	contractFunction, ok := contractMember.(interpreter.FunctionValue)
+	if !ok {
+		return interpreter.NotInvokableError{
+			Value: contractFunction,
+		}
+	}
+
+	_, err = inter.InvokeFunction(contractFunction, invocation)
+	if err != nil {
+		return newError(err, context)
+	}
+
+	return nil
```

Comment:
Maybe don't ignore the value returned from the function, but return it from this runtime function